types.go

```diff
@@ -24,6 +24,108 @@ func (t Timestamp) Parse() (time.Time, error) {
 	return time.Parse(time.RFC3339, string(t))
 }
 
+// NullString is a custom String type to allow for optional modifying strings that can be reset
+// the JSON Marshalling encoding is unique:
+// if the string is empty, it will be encoded as null
+// if the string is not empty, it will be encoded as a string
+type NullString string
+
+// MarshalJSON returns the JSON encoding of the NullString
+func (s NullString) MarshalJSON() ([]byte, error) {
+	if s == "" {
+		return json.Marshal(nil)
+	}
+
+	return json.Marshal(string(s))
+}
+
+func nullString(v interface{}) *NullString {
+	switch v.(type) {
+	case string:
+		if v.(string) == "" {
+			return nil
+		}
+
+		ns := NullString(v.(string))
+		return &ns
+	case NullString:
+		ns := NullString(v.(NullString))
+		return &ns
+	case *NullString:
+		return v.(*NullString)
+	default:
+		return nil
+	}
+}
+
+// NullBool is a custom Bool type to allow for optional modifying booleans that can have false as value
+// the JSON Marshalling encoding is unique:
+// if the bool is false, it will be encoded as false
+// if the bool is true, it will be encoded as true
+type NullBool bool
+
+// MarshalJSON returns the JSON encoding of the NullBool
+func (b NullBool) MarshalJSON() ([]byte, error) {
+	if b == false {
+		return json.Marshal(false)
+	}
+
+	return json.Marshal(true)
+}
+
+func nullBool(v interface{}) *NullBool {
+	switch v.(type) {
+	case bool:
+		if v.(bool) == false {
+			return nil
+		}
+
+		nb := NullBool(true)
+		return &nb
+	case NullBool:
+		nb := NullBool(v.(NullBool))
+		return &nb
+	case *NullBool:
+		return v.(*NullBool)
+	default:
+		return nil
+	}
+}
+
+// NullInt is a custom Int type to allow for optional modifying integers that can have 0 as value
+// the JSON Marshalling encoding is unique:
+// if the integer is 0, it will be encoded as 0
+// if the integer is not equal to 0, it will be encoded as an integer
+type NullInt int
+
+// MarshalJSON returns the JSON encoding of the NullBool
+func (i NullInt) MarshalJSON() ([]byte, error) {
+	if i == 0 {
+		return json.Marshal(0)
+	}
+
+	return json.Marshal(int(i))
+}
```

Comment:
If `i == 0`, should we marshal `nil`? If not, I'm not sure how this would operate differently from using a regular `int`.

```suggestion
func (i NullInt) MarshalJSON() ([]byte, error) {
	if i == 0 {
		return json.Marshal(nil)
	}

	return json.Marshal(i)
}
```